platform/x11/detect.py

```diff
@@ -122,26 +123,18 @@ def configure(env):
     # Cross-compilation
     # TODO: Support cross-compilation on architectures other than x86.
     host_is_64_bit = sys.maxsize > 2**32
+    if env["arch"] == "":
+        env["arch"] = detect_arch()
     if env["bits"] == "default":
-        env["bits"] = "64" if host_is_64_bit else "32"
-    if host_is_64_bit and (env["bits"] == "32" or env["arch"] == "x86"):
+        env["bits"] = "64" if env["arch"].endswith("64") else "32"
+
+    if host_is_64_bit and env["arch"] == "x86":
         env.Append(CCFLAGS=["-m32"])
         env.Append(LINKFLAGS=["-m32"])
```

Comment:
The `host_is_64_bit` check is not needed here. `-m32` specifies x86_32, and will still work if the host is x86_32. The important check here is the architecture of the target (note: not the bitness of the target).

Reply:
Ah, that's right! Thanks!

I've removed the host check and made it a target check.